spec/peep_dao_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'peep_dao'
+
+describe PeepDao do
+
+  let(:peep) { double :peep }
+  let(:peep_with_no_peep) { double :peep }
+  let(:time_now) { Time.now }
+
+  describe '#all' do
+    it 'should return an array of Peep objects' do
+      all_peeps = described_class.all
+      expect(all_peeps.class).to eq(Array)
+      all_peeps.each do |peep|
+        expect(peep.class).to eq(Peep)
+      end
+    end
+
+    it 'should return an array with the most recent posted peep first if reverse parameter passed in' do
+      allow(peep).to receive(:message).and_return("message")
+      allow(peep).to receive(:date).and_return(time_now + 100) ## adds 100 seconds to date of new post to make sure it is later
```

Comment:
Great use of doubles and allows throughout your unit tests! Great setup for each of your tests!